test/test_automated_install.py

```diff
@@ -120,101 +154,238 @@ def test_configureFirewall_no_firewall(Pihole):
     expected_stdout = 'No active firewall detected'
     assert expected_stdout in configureFirewall.stdout
 
+
 def test_configureFirewall_IPTables_enabled_declined_no_errors(Pihole):
-    ''' confirms IPTables rules are not applied when IPTables is running, user declines ruleset '''
+    '''
+    confirms IPTables rules are not applied when IPTables is running, user
+    declines ruleset
+    '''
     # iptables command exists
-    mock_command('iptables', {'*':('', '0')}, Pihole)
+    mock_command('iptables', {'*': ('', '0')}, Pihole)
     # modinfo returns always true (ip_tables module check)
-    mock_command('modinfo', {'*':('', '0')}, Pihole)
+    mock_command('modinfo', {'*': ('', '0')}, Pihole)
     # Whiptail dialog returns Cancel for user prompt
-    mock_command('whiptail', {'*':('', '1')}, Pihole)
+    mock_command('whiptail', {'*': ('', '1')}, Pihole)
     configureFirewall = Pihole.run('''
     source /opt/pihole/basic-install.sh
     configureFirewall
     ''')
     expected_stdout = 'Not installing firewall rulesets.'
     assert expected_stdout in configureFirewall.stdout
 
+
 def test_configureFirewall_IPTables_enabled_rules_exist_no_errors(Pihole):
-    ''' confirms IPTables rules are not applied when IPTables is running and rules exist '''
-    # iptables command exists and returns 0 on calls (should return 0 on iptables -C)
-    mock_command('iptables', {'-S':('-P INPUT DENY', '0')}, Pihole)
+    '''
+    confirms IPTables rules are not applied when IPTables is running and rules
+    exist
+    '''
+    # iptables command exists and returns 0 on calls
+    # (should return 0 on iptables -C)
+    mock_command('iptables', {'-S': ('-P INPUT DENY', '0')}, Pihole)
     # modinfo returns always true (ip_tables module check)
-    mock_command('modinfo', {'*':('', '0')}, Pihole)
+    mock_command('modinfo', {'*': ('', '0')}, Pihole)
     # Whiptail dialog returns Cancel for user prompt
-    mock_command('whiptail', {'*':('', '0')}, Pihole)
+    mock_command('whiptail', {'*': ('', '0')}, Pihole)
     configureFirewall = Pihole.run('''
     source /opt/pihole/basic-install.sh
     configureFirewall
     ''')
     expected_stdout = 'Installing new IPTables firewall rulesets'
     assert expected_stdout in configureFirewall.stdout
     firewall_calls = Pihole.run('cat /var/log/iptables').stdout
-    assert 'iptables -I INPUT 1 -p tcp -m tcp --dport 80 -j ACCEPT' not in firewall_calls
-    assert 'iptables -I INPUT 1 -p tcp -m tcp --dport 53 -j ACCEPT' not in firewall_calls
-    assert 'iptables -I INPUT 1 -p udp -m udp --dport 53 -j ACCEPT' not in firewall_calls
+    iptables_line = 'iptables -I INPUT 1 -p tcp -m tcp --dport 80 -j ACCEPT'
```

Comment:
This is a fragile test as is because of the looking for absence for very exact strings vs looser rules:

```
assert '-L' in firewall_calls
assert -I' not in firewall_calls
```

Could even count the number of IPTables calls to just be 1 if all we do is iptables -L to verify existing rules (using a `.run(wc -l)` or count of `\n` character using stdout perhaps.